docs/src/main/asciidoc/rest-data-panache.adoc

```diff
@@ -354,6 +354,42 @@ It applies to the paged resources only and is a number starting with 1. Default
 Fields are sorted in the ascending order unless they're prefixed with a `-`.
 E.g. `?sort=name,-age` will sort the result by the name ascending by the age descending.
 
+== Resource Method Before/After Listeners
+
+REST Data with Panache supports the subscription to the following resource method hooks: 
+
+* Before/After add resource
+* Before/After update resource
+* Before/After delete resource
+
+To register your resource method listener, you need to provide a bean that implements the interface `RestDataResourceMethodListener`, for example:
+
+[source,java]
+----
+@ApplicationScoped
+public class PeopleRestDataResourceMethodListener implements RestDataResourceMethodListener<Person> {
+    @Override
+    public void onBeforeAdd(Person person) {
+        System.out.println("Before Save Person: " + person.name);
+    }
+}
+----
+
+When using the REST Data Reactive with Panache extension, you will receive the Uni instance of the entity in the after methods:
+
+[source,java]
+----
+@ApplicationScoped
+public class PeopleRestDataResourceMethodListener implements RestDataResourceMethodListener<Person> {
+    @Override
+    public void onAfterAdd(Uni<Person> uni) {
+        uni.subscribe().with(person -> {
```

Comment:
I would like to know more about why this is `Uni` here. 
Furthermore, is this same `Uni` that is returned by HR? If so, what happens when something like RR subscribed to the same `Uni`?

Reply:
To be honest, this does not seem terribly user friendly, why would we not provide users the entity itself?

Reply:
Yeah I understand the reasoning. But it's not very user friend really - there is almost never a good reason to force users to subscribe to a `Uni`